src/ucp/rndv/rndv.c

```diff
@@ -507,14 +507,16 @@ ucp_rndv_progress_rma_zcopy_common(ucp_request_t *req, ucp_lane_index_t lane,
         }
 
         ucp_request_send_state_advance(req, &state, proto, status);
-        if (req->send.state.dt.offset == req->send.length) {
-            if (req->send.state.uct_comp.count == 0) {
-                uct_completion_update_status(&req->send.state.uct_comp, status);
-                req->send.state.uct_comp.func(&req->send.state.uct_comp);
+        if (ucs_likely(!UCS_STATUS_IS_ERR(status))) {
+            if (req->send.state.dt.offset == req->send.length) {
+                if (req->send.state.uct_comp.count == 0) {
+                    req->send.state.uct_comp.func(&req->send.state.uct_comp);
+                }
+                return UCS_OK;
             }
```

Comment:
similar to comment above it could be part of `ucp_request_send_state_advance`

Reply:
https://github.com/openucx/ucx/pull/6983#discussion_r656321978

Reply:
what about this?

Reply:
I'd keep it as is to align with other places, e.g. https://github.com/openucx/ucx/blob/4e41f50c9ce997d9b84b025ef0e8601c76734bf3/src/ucp/proto/proto_am.inl#L325

If we want to move the code above also to under `send_state_advance`, we need add `complete` parameter

Reply:
ok, actually it does not "advance" data type state, so lets keep it outside